ra2ce/analysis/adaptation/adaptation.py

```diff
@@ -0,0 +1,111 @@
+"""
+                    GNU GENERAL PUBLIC LICENSE
+                      Version 3, 29 June 2007
+
+    Risk Assessment and Adaptation for Critical Infrastructure (RA2CE).
+    Copyright (C) 2023 Stichting Deltares
+
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+from pathlib import Path
+
+from geopandas import GeoDataFrame
+
+from ra2ce.analysis.adaptation.adaptation_option import AdaptationOption
+from ra2ce.analysis.adaptation.adaptation_option_collection import (
+    AdaptationOptionCollection,
+)
+from ra2ce.analysis.analysis_config_data.analysis_config_data import (
+    AnalysisConfigData,
+    AnalysisSectionAdaptation,
+)
+from ra2ce.analysis.analysis_input_wrapper import AnalysisInputWrapper
+from ra2ce.analysis.damages.analysis_damages_protocol import AnalysisDamagesProtocol
+from ra2ce.network.graph_files.network_file import NetworkFile
+
+
+class Adaptation(AnalysisDamagesProtocol):
+    analysis: AnalysisSectionAdaptation
+    graph_file: NetworkFile
+    graph_file_hazard: NetworkFile
+    input_path: Path
+    output_path: Path
+    _adaptation_options: AdaptationOptionCollection
+
+    # TODO: add the proper protocol for the adaptation analysis.
+    def __init__(
+        self, analysis_input: AnalysisInputWrapper, analysis_config: AnalysisConfigData
+    ):
+        self.analysis = analysis_input.analysis
+        self._adaptation_options = AdaptationOptionCollection.from_config(
+            analysis_config
+        )
+
+    def execute(self) -> GeoDataFrame | None:
+        """
+        Run the adaptation analysis.
+        """
+        return self.calculate_bc_ratio()
+
+    def calculate_option_cost(self, option: AdaptationOption) -> float:
```

Comment:
Shouldn't  it a method of the class AdaptationOption?

Reply:
Add in docstrings: calculate the net value unit cost (per m)

Reply:
Fair point, will adjustust